coderd/rbac/regosql/configs.go

```diff
@@ -2,6 +2,10 @@ package regosql
 
 import "github.com/coder/coder/coderd/rbac/regosql/sqltypes"
 
+func resourceIDMatcher() sqltypes.VariableMatcher {
+	return sqltypes.StringVarMatcher("id :: text", []string{"input", "object", "id"})
```

Comment:
It is a consequence of not being able to know if a input is a uuid or a string. Since we are converting from rego to sql, the rego values are `strings`. There is no concept of a `uuid` in our rego policy.

So when I do the conversion, there is the option to convert the `[]string{}` to `[]uuid{}` or convert the variable `input.object.id` to `:: text`.

You have to convert one side of the comparison. Since we have to interpret variables anyway, it felt more natural to typecast `input.object.id` then to somehow understand which `[]string{}` are `uuid` and which are `[]string{}`.

----

So doing it this way actually simplifies our conversions by never dealing with `uuid` type in the converter.

You can see that we need to typecast one in this db fiddle: https://www.db-fiddle.com/f/qb4U7YPfJNju4dBFMhr1Bz/1